### DISCUSSION TITLE
Why can't I find the libraries I already installed？

### DISCUSSION THREAD

**Discussion by Luo-Chenxin** [microsoft/vcpkg#27946] [General]
I solved this problem by adding the "CMAKE_TOOLCHAIN_FILE" and "VCPKG_TARGET_TRIPLET" parameters in setting.json instead of in the settings.
But there is a new problem in cmake.
> [main] Building folder: LD_paser_linux 
[main] Configuring folder: LD_paser_linux 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_TOOLCHAIN_FILE:STRING=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET:STRING=x64-windows -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -Sd:/workfile2/LD/demo/LD_paser_linux -Bd:/workfile2/LD/demo/LD_paser_linux/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] CMake Error at D:/vcpkg/scripts/buildsystems/vcpkg.cmake:834 (_find_package):
[cmake]   Could not find a configuration file for package "jsoncpp" that is
[cmake]   compatible with requested version "".
[cmake] 
[cmake]   The following configuration files were considered but not accepted:
[cmake] 
[cmake]     D:/vcpkg/installed/x64-windows/share/jsoncpp/jsoncppConfig.cmake, version: 1.9.5 (64bit)
[cmake] 
[cmake] Call Stack (most recent call first):
[cmake]   CMakeLists.txt:11 (find_package)
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[cmake] See also "D:/workfile2/LD/demo/LD_paser_linux/build/CMakeFiles/CMakeOutput.log".
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_TOOLCHAIN_FILE:STRING=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET:STRING=x64-windows -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -Sd:/workfile2/LD/demo/LD_paser_linux -Bd:/workfile2/LD/demo/LD_paser_linux/build -G "Unix Makefiles" exited with code: 1 and signal: null

**Answer:** The compiler architecture triplet and library do not match, and just replace them with the same version.

**Original Question:**
I want to use jsoncpp as a third-party library in my project，and I have successfully installed it using vcpkg.
![5](https://user-images.githubusercontent.com/51735545/203191221-0a3060dd-8af4-4d68-86ab-f3063b7bcce3.PNG)
Then I tried using find_package() in CMakeList to find it but failed.
![7](https://user-images.githubusercontent.com/51735545/203191577-6c415a33-a082-4de9-b350-f7218d883283.PNG)
Of course I have specified the CMake parameters that I consider appropriate.
![6](https://user-images.githubusercontent.com/51735545/203191854-25333d3b-f10a-4328-ab78-76b6b63a3f22.PNG)
development environment：Windows , Visual Studio code

**Comment by JackBoosY** [comment]
> I solved this problem by adding the "CMAKE_TOOLCHAIN_FILE" and "VCPKG_TARGET_TRIPLET" parameters in setting.json instead of in the settings.

I think you should clean up the cmake cache.

Can you double confirm that your custom project configuration is `x64` and dynamic?
And please provide the repro step and code.

**Reply by Luo-Chenxin** [reply]
I'm sorry. For some reasons, this project gave up using vckpg for package management, so I'm afraid I can't go back to the original state.
